vowpalwabbit/text_parser/include/vw/text_parser/parse_example_text.h

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) by respective owners including Yahoo!, Microsoft, and
+// individual contributors. All rights reserved. Released under a BSD (revised)
+// license as described in the file LICENSE.
+#pragma once
+
+#include "vw/common/string_view.h"
+#include "vw/core/multi_ex.h"
+#include "vw/core/vw_fwd.h"
+
+#include <cstdint>
+
+namespace VW
+{
+namespace parsers
+{
+namespace text
+{
+void substring_to_example(VW::workspace* all, VW::example* ae, VW::string_view example);
+
+void read_line(VW::workspace& all, example* ex, VW::string_view line);  // read example from the line.
```

Comment:
Whats the difference between `substring_to_example` and `read_line`? Can we consolidate?

Reply:
`read_line` reads until a newline is found and then passes the string to `substring_to_example`

`substring_to_example` is used by the throughput tool and internally in other parser methods, could probably just be made `details` and only expose `read_line`